pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -576,11 +577,23 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 
 		// If the pod doesn't exist yet, create it
 		if pod == nil {
+			transferNetwork, found, err := unstructured.NestedFieldNoCopy(crInstance.Object, "spec", "transferNetwork")
```

Comment:
why "spec" is needed?

Reply:
this is how it's often used, probably just to ensure the nested field has the right parent. But I am changing it to NestedStringMap to avoid casting

Reply:
actually i tried to do it without it for a PR i have, it didn't work. but when pointing out the tree it does...

Reply:
even better :)